impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java

```diff
@@ -230,18 +228,45 @@ public ConfigHolder generateConfigForSpecificNode(KafkaEndpoints kafkaEndpoints,
         var earlyStart = new TreeSet<String>();
 
         final ConfigHolder configHolder;
-        if (role.contains(BROKER_ROLE)) {
-            configHolder = configureBroker(kafkaEndpoints, nodeId, protocolMap, listeners, advertisedListeners, earlyStart, nodeConfiguration);
+        if (NodeRole.hasBrokerRole(roles)) {
+            var interBrokerEndpoint = kafkaEndpoints.getEndpointPair(Listener.INTERNAL, nodeId);
```

Comment:
Broker and controller are non-exclusive roles which a node may have. Having separate methods for broker and controller obscures the combined case. Certainly we can try to tidy this up a bit, but I think it would be better to move the ConfigHolder instantiation outside of the if/else, to make it clear that it's only the endpoints which are different when a node is a broker.